### PR TITLE
MINOR: some more Javadocs

### PR DESCRIPTION
**Pull request by guozhangwang**

### CODE REVIEW

**Comment by guozhangwang** [comment]
ping @ymatsuda @ewencp for reviews.

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java [hunk 1]
```diff
@@ -29,7 +29,8 @@
 import java.util.concurrent.atomic.AtomicInteger;
 
 /**
- * KStreamBuilder is the class to create KStream instances.
+ * KStreamBuilder is a superclass of {@link TopologyBuilder} that provides the {@link KStream} DSL
```
Comment (ymatsuda) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java]: a subclass

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java [hunk 2]
```diff
@@ -115,6 +117,13 @@ public KStreamBuilder() {
         return KStreamImpl.merge(this, streams);
     }
 
+    /**
+     * Create a unqiue processor name used for translation into the processor topology.
+     * This function should only be used for users to add their own DSL operators and corresponding translation rules.
```
Comment (ymatsuda) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java]: Isn't this misleading? It sounds like the framework is user-extensible.

**Comment by ymatsuda** [comment]
I left a few comments. Otherwise, LGTM.

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java (hunk 2 above)
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java]: @ymatsuda I was thinking that developers could add their own operators with the translation rules, which hence needs to call this function. Maybe I was thinking too much as there won't be many people doing it :). May be just change to "This function is for internal usage only?".

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java (hunk 1 above)
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java]: Ack.

File: streams/src/main/java/org/apache/kafka/streams/kstream/Reducer.java [hunk 3]
```diff
@@ -17,6 +17,11 @@
 
 package org.apache.kafka.streams.kstream;
 
+/**
+ * The Reducer interface for combinging two values of the same type into a new value.
```
Comment (ewencp) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Reducer.java]: typo: combinging

Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Reducer.java]: Ack.

**Comment by ewencp** [comment]
LGTM

**Pull request closed without merging**